File: oa-portal/problems/a1-frog-1/starters/main.cpp
#include <bits/stdc++.h>
using namespace std;

// h is 1-indexed. From stone i the frog jumps to i+1 or i+2, paying
// |h[i] - h[j]|. Minimum total cost to reach stone n.
long long minCost(const vector<long long>& h) {
    int n = (int)h.size() - 1;
    // write your code here

    (void)n;
    return 0;
}

int main() {
    int n;
    if (scanf("%d", &n) != 1) return 0;
    vector<long long> h(n + 1);
    for (int i = 1; i <= n; i++) scanf("%lld", &h[i]);

    printf("%lld\n", minCost(h));
    return 0;
}

File: oa-portal/problems/a1-frog-1/starters/main.java
import java.io.*;
import java.util.*;

public class Main {

    // h is 1-indexed (h[0] unused). From stone i the frog jumps to i+1 or i+2,
    // paying |h[i] - h[j]|. Minimum total cost to reach stone n.
    static long minCost(long[] h) {
        // write your code here

        return 0;
    }

    public static void main(String[] args) throws IOException {
        FastReader in = new FastReader();
        int n = in.nextInt();
        long[] h = new long[n + 1];
        for (int i = 1; i <= n; i++) h[i] = in.nextLong();

        System.out.println(minCost(h));
    }

    static class FastReader {
        private final DataInputStream in = new DataInputStream(new BufferedInputStream(System.in, 1 << 16));
        int nextInt() throws IOException { return (int) nextLong(); }
        long nextLong() throws IOException {
            int b = in.read();
            while (b != '-' && (b < '0' || b > '9')) b = in.read();
            boolean neg = b == '-';
            if (neg) b = in.read();
            long v = 0;
            while (b >= '0' && b <= '9') { v = v * 10 + (b - '0'); b = in.read(); }
            return neg ? -v : v;
        }
        String next() throws IOException {
            int b = in.read();
            while (b <= ' ') b = in.read();
            StringBuilder sb = new StringBuilder();
            while (b > ' ') { sb.append((char) b); b = in.read(); }
            return sb.toString();
        }
    }
}
